### PR TITLE
feat: Add scoped self-signed JWT support for REST

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: Google.Api.Gax.Rest/ClientBuilderBase.cs [hunk 1]
```diff
@@ -99,6 +99,15 @@ public abstract class ClientBuilderBase<TClient>
         /// </remarks>
         public IHttpClientFactory HttpClientFactory { get; set; }
 
+        /// <summary>
+        /// Returns whether or not self-signed JWTs will be used over OAuth tokens when OAuth scopes are explicitly set.
```
Comment (amanda-tarafa) [inline Google.Api.Gax.Rest/ClientBuilderBase.cs:103]: Do we have a note about this only applying to SAs in GAPICs? If no, then that's fine.

File: Google.Api.Gax.Rest.Tests/ClientBuilderBaseTest.GetHttpClientInitializerTest.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            {
+                var path = Path.GetTempFileName();
+                File.WriteAllText(path, TestServiceAccountJson);
+
+                var builder = new FakeBuilder(s_scopes, true)
+                {
+                    CredentialsPath = path
+                };
+                await AssertCredentialAsync(builder, AssertBasicProperties);
+                File.Delete(path);
```
Comment (amanda-tarafa) [inline Google.Api.Gax.Rest.Tests/ClientBuilderBaseTest.GetHttpClientInitializerTest.cs:52]: Do you have to delete it if it's a temp file?
I never do that, as I think it deletes on its own when it reaches a threshold.
Fine by me if this is a precaution though.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

File: Google.Api.Gax.Rest/ClientBuilderBase.cs (hunk 1 above)
Comment (jskeet) [reply Google.Api.Gax.Rest/ClientBuilderBase.cs:103]: This comment is directly copy/pasted from the gRPC one :)

File: Google.Api.Gax.Rest.Tests/ClientBuilderBaseTest.GetHttpClientInitializerTest.cs (hunk 2 above)
Comment (jskeet) [reply Google.Api.Gax.Rest.Tests/ClientBuilderBaseTest.GetHttpClientInitializerTest.cs:52]: We probably don't *have* to, but we might as well. This whole test is copy/pasted from gRPC and changed just enough to make it pass (with the slightly different API we expose).

**Pull request merged**